js/components/urlBar.js

```diff
@@ -207,6 +222,28 @@ class UrlBar extends ImmutableComponent {
     windowActions.setUrlBarSelected(false)
     windowActions.setUrlBarActive(true)
     windowActions.setNavBarUserInput(e.target.value)
+    let location = this.props.urlbar.get('location')
+    if (location !== null && location.length !== 0) {
+      const isLocationUrl = isUrl(location)
+      if (!isLocationUrl) {
+        let entries = searchProviders.providers
+        entries.forEach((entry) => {
+          const searchRE = new RegExp('^' + entry.shortcut + ' .*', 'g')
+          if (searchRE.test(location)) {
+            const iconSize = 16
+            this.searchFaviconStyle = {
```

Comment:
Style should really only be calculated inside render, here you'd set state which indicated the active engine, something like `frame.x.navbar.urlbar.activeSearchEngine`.   I think we should also show the favicon as soon as you put searchTerm + ' ' and you don't need an extra char after that. I think it will be fixed by what I'm suggesting here.  
You might want to wait for Brad to weigh in first to make sure we still want to have the favicon in the urlbar because I mentioned some of the favicons are looking bad as is, like yahoo and duck duck go seems pixelated.